https://bugs.gentoo.org/885763
https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1114
https://gitlab.freedesktop.org/xorg/xserver/-/commit/34a430a16ec2cb9851b1f2bc7fc2117e19b9fcc7

From 34a430a16ec2cb9851b1f2bc7fc2117e19b9fcc7 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Thu, 13 Apr 2023 15:45:58 +0200
Subject: [PATCH] fb: Declare wfbFinishScreenInit, wfbScreenInit for
 !FB_ACCESS_WRAPPER

xf86-video-nouveau calls wfbScreenInit without defining
FB_ACCESS_WRAPPER (which has other unintended side effects).
Presently, this compiles and links because compilers still support
implicit function declarations, but this is going to change fairly
soon.  This seems to be the most straightforward change to keep
the driver building.
---
 fb/fb.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/fb/fb.h b/fb/fb.h
index 08143a0d49..309f15294c 100644
--- a/fb/fb.h
+++ b/fb/fb.h
@@ -1027,7 +1027,6 @@ extern _X_EXPORT Bool
                int dpiy, int width,     /* pixel width of frame buffer */
                int bpp);        /* bits per pixel of frame buffer */
 
-#ifdef FB_ACCESS_WRAPPER
 extern _X_EXPORT Bool
 wfbFinishScreenInit(ScreenPtr pScreen,
                     void *pbits,
@@ -1049,7 +1048,6 @@ wfbScreenInit(ScreenPtr pScreen,
               int width,
               int bpp,
               SetupWrapProcPtr setupWrap, FinishWrapProcPtr finishWrap);
-#endif
 
 extern _X_EXPORT Bool
 fbFinishScreenInit(ScreenPtr pScreen,
-- 
GitLab

